src/command_modules/azure-cli-backup/azure/cli/command_modules/backup/custom.py

```diff
@@ -610,6 +621,23 @@ def _get_resource_name_and_rg(resource_group_name, name_or_id):
     return name, resource_group
 
 
+def _validate_container(container):
+    _validate_object(container, "Container not found. Please provide a valid container_name.")
+
+
+def _validate_item(item):
+    _validate_object(item, "Item not found. Please provide a valid item_name.")
+
+
+def _validate_policy(policy):
+    _validate_object(policy, "Policy not found. Please provide a valid policy_name.")
```

Comment:
It doesn't look like this addresses the key parts of #4502.

> In this case, it was not clear to me that a container and item had been created by the enable-for-vm command, so I assumed that these names were the names that would be created. 

#4502 calls for this to distinction to be made.

And then...

> The parameter should definitely accept the format of name that the container object actually uses.

I don't see that this change does that.  If the user programmatically obtains the name in the native format, the command should not choke on that.  The behavior should be similar to other commands in the CLI that accept names or IDs.  If it detects that the user supplied the long form of the name, it should parse that long form into the individual name parameters.  This scenario should also definitely be covered by a test.

Reply:
Yes, @tjprescott , as I said earlier, this change partially addresses #4502. Updated the history to reflect the correct changes. Will roll out the complete fix for #4502 in the next release.